### PR TITLE
Infer daml-script package id from script

### PR DESCRIPTION
**Pull request by aherrmann-da**
Similar to https://github.com/digital-asset/daml/pull/5016 but for daml-script this detects the package id of the daml-script library based on the type of the script we are running.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala [hunk 1]
```diff
@@ -107,6 +107,47 @@ object ParticipantsJsonProtocol extends DefaultJsonProtocol {
...
+    fromIdentifier(compiledPackages, scriptId)
+  }
+  def fromIdentifier(compiledPackages: CompiledPackages, scriptId: Identifier): Script = {
+    val scriptExpr = SEVal(LfDefRef(scriptId), None)
+    val scriptTy = compiledPackages
+      .getPackage(scriptId.packageId)
+      .flatMap(_.lookupIdentifier(scriptId.qualifiedName).toOption) match {
+      case Some(DValue(ty, _, _, _)) => ty
+      case Some(d @ DTypeSyn(_, _)) =>
+        throw new RuntimeException(s"Expected DAML script but got synonym $d")
```
Comment (cocreature) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala]: It probably makes sense to also change this to return `Either[String, Script]` similar to the trigger runner.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/RunnerMain.scala [hunk 2]
```diff
@@ -84,8 +84,15 @@ object RunnerMain {
...
+        val script = try {
+          Script.fromDar(dar, scriptId)
+        } catch {
+          case e: Throwable =>
+            system.terminate
+            sys.exit(1)
+        }
         val runner = try {
-          new Runner(dar, applicationId, commandUpdater, timeProvider)
+          new Runner(dar, script.scriptIds, applicationId, commandUpdater, timeProvider)
```
Comment (cocreature) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/RunnerMain.scala]: Assuming I understand your plans for the follow-up PRs correctly, in the end the constructor shouldn’t throw anymore so we can probably remove the try here. Nothing to do in this PR though, just a note for the future :slightly_smiling_face:

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/TestMain.scala [hunk 3]
```diff
@@ -123,29 +122,46 @@ object TestMain extends StrictLogging {
           success = new AtomicBoolean(true)
           _ <- Future.sequence {
             dar.main._2.modules.flatMap {
-              case (moduleName, module) =>
+              case (moduleName, module) => {
+                object ScriptDefinition {
```
Comment (cocreature) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/TestMain.scala]: Maybe worth factoring this out to a top-level definition? I’m also not a huge fan of `unapply`, I always find it a bit hart to read and iirc it also breaks completeness checks. Can we just separate this in two steps: First use `collect` as a `mapMaybe` where the function is something like the current `unapply` (you can turn a function returning an optional into a partial function using `Function.unlift` I believe or just define it directly as a partial function). Then we have a list of all scripts. And then in a second step run them.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala (hunk 1 above)
Comment (aherrmann-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala]: Yes, it turns out this allows to simplify the `unapply` bit in `TestMain`. I've updated the PR.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/RunnerMain.scala (hunk 2 above)
Comment (aherrmann-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/RunnerMain.scala]: Agreed.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/TestMain.scala (hunk 3 above)
Comment (aherrmann-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/TestMain.scala]: `Function.unlift` is a good hint, thanks! It turns out that making `Script.fromIdentifier|Dar` pure allows to simplify this bit. I've pulled it out into a separate step that collects all test scripts that take no arguments into a list using `collect(Function.unlift ...)`

**Pull request merged**